packages/playwright-core/src/server/browserContext.ts

```diff
@@ -380,6 +379,26 @@ export abstract class BrowserContext extends SdkObject {
     await this.doClearPermissions();
   }
 
+  async setExtraHTTPHeaders(progress: Progress, headers: types.HeadersArray) {
+    const oldHeaders = this._options.extraHTTPHeaders;
+    this._options.extraHTTPHeaders = headers;
+    progress.cleanupWhenAborted(async () => {
+      this._options.extraHTTPHeaders = oldHeaders;
+      await this.doUpdateExtraHTTPHeaders();
```

Comment:
Nit: maybe just pass the new headers as a param and only update them in options on success?

Reply:
That's an interesting idea, but makes it harder to wire up on the browser-specific implementations.